src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -247,6 +257,24 @@ public void CreateValidationMetadata(ValidationMetadataProviderContext context)
             }
         }
 
+        private Func<string> LocalizeAttributeProperty(Func<string> GetProperty, IStringLocalizer localizer)
+        {
+            return () =>
+            {
+                var value = GetProperty();
+                if (!string.IsNullOrEmpty(value) && localizer != null)
+                {
+                    var localizedValue = localizer[value];
+                    if (value != localizedValue)
+                    {
+                        value = localizedValue;
+                    }
+                }
+
+                return value;
+            };
```

Comment:
Rewrite this to do the `null` check on `localizer` up front, returning `propertyGetter` in the simple and common case that `localizer` is `null`.